pkgs/by-name/bi/bitwarden-desktop/package.nix

```
@@ -75,32 +79,39 @@ buildNpmPackage rec {
   npmDepsHash = "sha256-fYZJA6qV3mqxO2g+yxD0MWWQc9QYmdWJ7O7Vf88Qpbs=";
 
   cargoDeps = rustPlatform.fetchCargoVendor {
-    inherit pname version src;
-    patches = map (
-      patch:
-      runCommand (builtins.baseNameOf patch) { nativeBuildInputs = [ patchutils_0_4_2 ]; } ''
-        < ${patch} filterdiff -p1 --include=${lib.escapeShellArg cargoRoot}'/*' > $out
-      ''
-    ) patches;
-    patchFlags = [ "-p4" ];
-    sourceRoot = "${src.name}/${cargoRoot}";
+    inherit
+      pname
+      version
+      src
+      cargoRoot
+      patches
+      ;
     hash = "sha256-OldVFMI+rcGAbpDg7pHu/Lqbw5I6/+oXULteQ9mXiFc=";
   };
   cargoRoot = "apps/desktop/desktop_native";
 
   env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
 
-  nativeBuildInputs = [
-    cargo
-    copyDesktopItems
-    jq
-    makeWrapper
-    napi-rs-cli
-    pkg-config
-    rustc
-    rustPlatform.cargoCheckHook
-    rustPlatform.cargoSetupHook
-  ];
+  # make electron-builder not attempt to codesign the app on darwin
+  env.CSC_IDENTITY_AUTO_DISCOVERY = "false";
+
+  nativeBuildInputs =
+    [
+      cargo
+      jq
+      makeWrapper
+      napi-rs-cli
+      pkg-config
+      rustc
+      rustPlatform.cargoCheckHook
+      rustPlatform.cargoSetupHook
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      copyDesktopItems
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      xcbuild
```

```suggestion
      xcbuild
      darwin.autoSignDarwinBinariesHook
```

aarch64-darwin requires binaries to be signed.